### PR TITLE
TypeScript configuration

### PR DESCRIPTION
**Pull request by lynnagara**
The PR contains the commits from https://github.com/getsentry/sentry/pull/13786 related to supporting TypeScript configuration. Since there are many changes currently underway that change the build process for Sentry's UI, we should merge this first to avoid conflicts later.

### CODE REVIEW

**Review by dashed — APPROVED** [review]
whoot!  🚀

**Review by HazAT — APPROVED** [review]
:shipit:

File: package.json [hunk 1]
```diff
@@ -17,8 +17,11 @@
     "@babel/preset-env": "^7.0.0",
     "@babel/preset-react": "^7.0.0",
     "@babel/runtime": "^7.0.0",
-    "@sentry/browser": "5.2.0-beta.0",
-    "@sentry/integrations": "5.2.0-beta.0",
+    "@sentry/browser": "^5.4.2",
+    "@sentry/integrations": "^5.4.2",
+    "@sentry/typescript": "^5.3.0",
+    "@types/lodash": "^4.14.134",
```
Comment (billyvg) [inline package.json]: Should these be dev deps?

Comment (lynnagara) [reply package.json]: I think these are needed for the production build right now, but we should probably look into that. @HazAT - I know you mentioned something about this earlier, is that accurate?

**Pull request merged**